File: ingest-rest/src/main/java/org/chronopolis/ingest/api/serializer/DepositorContactSerializer.java
package org.chronopolis.ingest.api.serializer;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;
import org.chronopolis.rest.entities.DepositorContact;
import org.chronopolis.rest.models.DepositorContactModel;

import java.io.IOException;

/**
 * Serialize a DepositorContact entity to DepositorContactModel
 *
 * @author shake
 */
public class DepositorContactSerializer extends JsonSerializer<DepositorContact> {
    @Override
    public void serialize(DepositorContact contact,
                          JsonGenerator gen,
                          SerializerProvider serializers) throws IOException {
        DepositorContactModel model = new DepositorContactModel(contact.getContactName(),
                contact.getContactEmail(),
                contact.getContactPhone());
        gen.writeObject(model);
    }
}

File: rest-entities/src/main/kotlin/org/chronopolis/rest/entities/serializers/DepositorContactSerializer.kt
package org.chronopolis.rest.entities.serializers

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider
import org.chronopolis.rest.entities.depositor.DepositorContact as DepositorContactEntity

/**
 * Serializer to return a [DepositorContact] model from an [DepositorContactEntity] entity
 *
 * @author shake
 */
class DepositorContactSerializer : JsonSerializer<DepositorContactEntity>() {
    override fun serialize(contact: DepositorContactEntity,
                           gen: JsonGenerator,
                           serializers: SerializerProvider) {
        gen.writeObject(contact.model())
    }
}